### PR TITLE
support parameter

### PR DESCRIPTION
**Pull request by czpmango**

### CODE REVIEW

File: nebula2/gclient/net/Connection.py [hunk 1]
```diff
@@ -105,6 +105,28 @@ def execute(self, session_id, stmt):
                     raise IOErrorException(IOErrorException.E_UNKNOWN, te.message);
             raise
 
+    def execute_parameter(self, session_id, stmt, params):
+        """execute interface with session_id and ngql
+        :param session_id: the session id get from result of authenticate interface
+        :param stmt: the ngql
+        :return: ExecutionResponse
```
Comment (jievince) [inline nebula2/gclient/net/Connection.py:112]: add notes for params

File: nebula2/gclient/net/Session.py [hunk 2]
```diff
@@ -60,6 +60,37 @@ def execute(self, stmt):
         except Exception:
             raise
 
+    def execute_parameter(self, stmt, params):
+        """execute statement
+
+        :param stmt: the ngql
+        :return: ResultSet
```
Comment (jievince) [inline nebula2/gclient/net/Session.py:67]: add notes for params

File: nebula2/gclient/net/Connection.py (hunk 1 above)
Comment (czpmango) [reply nebula2/gclient/net/Connection.py:113]: ACK.

File: nebula2/gclient/net/Session.py (hunk 2 above)
Comment (czpmango) [reply nebula2/gclient/net/Session.py:68]: Fixed.

**Review by jievince — APPROVED** [review]
LGTM

File: nebula2/gclient/net/Connection.py [hunk 3]
```diff
@@ -128,6 +151,29 @@ def execute_json(self, session_id, stmt):
...
+                if te.message.find("timed out") > 0:
+                    self._reopen()
+                    raise IOErrorException(IOErrorException.E_TIMEOUT, te.message)
+                elif te.type == TTransportException.END_OF_FILE:
+                    raise IOErrorException(IOErrorException.E_CONNECT_BROKEN, te.message)
+                elif te.type == TTransportException.NOT_OPEN:
+                    raise IOErrorException(IOErrorException.E_NOT_OPEN, te.message)
+                else:
+                    raise IOErrorException(IOErrorException.E_UNKNOWN, te.message);
+            raise
```
Comment (yixinglu) [inline nebula2/gclient/net/Connection.py:175]: let non-parameter interface reuse the parameter interface.

Comment (czpmango) [reply nebula2/gclient/net/Connection.py:175]: Fixed

**Review by yixinglu — APPROVED** [review]

**Pull request merged**